package/base-files/files/etc/init.d/led

```diff
@@ -3,6 +3,66 @@
 
 START=96
 
+led_color_set() {
+	local cfg="$1"
+	local sysfs="$2"
+
+	local max_b
+	local color_red
+	local color_green
+	local color_blue
+	local colors
+	local color
+	local write
+	local value
+
+	[ -e /sys/class/leds/${sysfs}/multi_intensity ] || return
+	[ -e /sys/class/leds/${sysfs}/multi_index ] || return
+
+	max_b="$(cat /sys/class/leds/${sysfs}/max_brightness)"
+
+	config_get color_red $1 color_red
+	config_get color_green $1 color_green
+	config_get color_blue $1 color_blue
+
+	write=0
+	if [ -n "$color_red" ]; then
+		[ "$color_red" -gt "$max_b" ] && color_red=$max_b
+		write=1
+	else
+		color_red=0
+	fi
+
+	if [ -n "$color_green" ]; then
+		[ "$color_green" -gt "$max_b" ] && color_green=$max_b
+		write=1
+	else
+		color_green=0
+	fi
+
+	if [ -n "$color_blue" ]; then
+		[ "$color_blue" -gt "$max_b" ] && color_blue=$max_b
+		write=1
+	else
+		color_blue=0
+	fi
+
+	[ "$write" = 1 ] || return
+
+	value=""
+	colors="$(cat /sys/class/leds/${sysfs}/multi_index | tr " " "\n")"
+	for color in $colors; do
```

Comment:
Hi! instead of hard-coding red, green and blue and duplicating all the code three times, you can loop over the available colors in multi_index and do something like:
```
config_get color_value $1 color_${color}
```
In this way, you reduced the code duplication and it's more generic for non-rgb leds.

Reply:
@981213 Thanks for your review and your suggestion. I have incorporated your changes. Please have a look on my pushed force branch.